dirsrvtests/tests/perf/ndncache_test.py

```diff
@@ -0,0 +1,577 @@
+                        res = re.match(r'.*optime=(\d+\.?\d*).*', line)
+                        if res:
+                            data.append(float(res.group(1)))
+                post_stats = self.getndncache_stats()
+                for k,v in post_stats.items():
+                    aggregated_stats[k] += float(v) - float(pre_stats[k])
+                self.postop()
+            result = Scenario.average(data)
+        except ldap.LDAPError as exc:
+            log.error(f'Scenario {scen} failed because of {exc}')
+            result = [ 0, 0, [] ]
+        log.info(f'result (average, normalized standard deviation, data) of {name} is {result}')
+        res_stats = [ str(aggregated_stats[a]) for a in STAT_ATTRS ]
+        res_stats2 = [ post_stats[a] for a in STAT_ATTRS2 ]
+        result.append(res_stats + res_stats2)
+        return result
+
+
+class Scen1(Scenario):
+    def __init__(self):
+        super().__init__()
+        self._name = 'scen1'
+        self._desc = 'Subtree search'
+
+    def op(self):
+        basedn = f'ou=Technology,ou=people,{DEFAULT_SUFFIX}'
+        result = self.ldc.search_ext_s(
+            base=basedn,
+            scope=ldap.SCOPE_SUBTREE,
+            filterstr='(uid=*)',
+            attrlist=['dn']
+        )
+
+
+class Scen2(Scenario):
+    def __init__(self):
+        super().__init__()
+        self.sss_control = SSSRequestControl(criticality=True, ordering_rules=['modifiersName'])
+        self._name = 'scen2'
+        self._desc = 'Server Side Sorted Subtree search'
+
+    def op(self):
+        basedn = f'ou=Technology,ou=people,{DEFAULT_SUFFIX}'
+        result = self.ldc.search_ext_s(
+            base=basedn,
+            scope=ldap.SCOPE_SUBTREE,
+            filterstr='(uid=*)',
+            serverctrls=[self.sss_control],
+            attrlist=['dn']
+        )
+
+
+class Scen3(Scenario):
+    def __init__(self):
+        super().__init__()
+        self._name = 'scen3'
+        self._desc = 'Subtree search with filter equality on dn syntax attribute'
+
+    def op(self):
+        basedn = f'ou=people,{DEFAULT_SUFFIX}'
+        filter = 'modifiersName=uid=bmcdonald,ou=Information Technology,ou=people,dc=example,dc=com'
+        result = self.ldc.search_ext_s(
+            base=basedn,
+            scope=ldap.SCOPE_SUBTREE,
+            filterstr=filter,
+            attrlist=['dn']
+        )
+
+
+class Scen4(Scenario):
+    def __init__(self):
+        super().__init__()
+        self._name = 'scen4'
+        self._desc = 'Modify member of large group with substring index'
+
+    def op(self):
+        basedn = f'cn=all_users,ou=groups,{DEFAULT_SUFFIX}'
+        value = b'uid=pwynn,ou=Information Technology,ou=people,dc=example,dc=com'
+        mods = [ ( ldap.MOD_DELETE, 'uniqueMember', [value,] ), ]
+        result = self.ldc.modify_s(basedn, mods)
+        mods = [ ( ldap.MOD_ADD, 'uniqueMember', [value,] ), ]
+        result = self.ldc.modify_s(basedn, mods)
```

Comment:
Scen4 takes a lot of time when I execute in `batch` mode...
Large group + substring index expensive rebuild on the large group + 200 index rebuilds for 100 measures. Do I understand it correctly?

I don't have a hard opinion here and it might be okay and expected to wait for the powerful test case.
I wonder if it'll be okay to make a custom number of measures or a smaller group here. WDYT?